exercise-app.js

```diff
@@ -28,13 +19,59 @@ const Weather = ({ temp, description, icon }) => (
 );
 
 
-const App = () => (
+class App extends React.Component {
+
+  constructor(props) {                // where do we hook up these props to ? where does this prop data come from?
+  super(props);                       // what is the purpose of this? why do we need to call super?
+  this.state = {
+  temp: 0,
+  description: 'not loaded yet',
+  icon: `http://openweathermap.org/img/w/10d.png`};  // always intialize a state? is the state to be updated by passing in props?
+  // this.reshapeData = this.reshapeData.bind(this)  // when do we need to bind and when not (I've seen both...)
```

Comment:
Yea, this one is confusing. It depends on how your project is setup. If your project was setup using Create React App or using the `babel-preset-react` then you do not need to bind. This is because both of those setups use `babel-plugin-transform-class-properties` which does the binding for you automatically. In this case I believe we are using `babel-preset-react`.